docs/design/ovs-pipeline.md

```diff
@@ -1,1197 +1,1938 @@
+into `EndpointIPField` and `EndpointPortField`, respectively. These loaded values will be consumed in table
+[EndpointDNAT] to which the packets are forwarded and in which DNAT will be performed. `RemoteEndpointRegMark` is loaded
+for remote Endpoints, such as in bucket 1.
 
-```text
-1. table=60, priority=200,ip,nw_src=10.10.1.2 actions=drop
-2. table=60, priority=200,ip,nw_src=10.10.1.3 actions=drop
-3. table=60, priority=0 actions=goto_table:61
-```
+The third group with `group_id` 11 is the destination of packets matched by flow 3, designed for a Service that has
+Endpoints and is configured with session affinity. The group closely resembles group 2, except that the destination of
+the packets is table [ServiceLB], rather than table [EndpointDNAT]. After being sent back to table [ServiceLB], they
+will be matched by flow 4.
 
-This table is also used to implement Antrea-native policy egress rules that are
-created in the Baseline Tier. Since the Baseline Tier is meant to be enforced
-after K8s NetworkPolicies, the corresponding flows will be created at a lower
-priority than K8s default drop flows. For example, a baseline rule to drop
-egress traffic to 10.0.10.0/24 for a Namespace will look like the following:
+### EndpointDNAT
+
+The table implements DNAT for Service connection after Endpoint selection is performed in table [ServiceLB].
+
+If you dump the flows of this table, you may see the following::
 
 ```text
-1. table=60, priority=80,ip,nw_src=10.10.1.11 actions=conjunction(5,1/2)
-2. table=60, priority=80,ip,nw_src=10.10.1.10 actions=conjunction(5,1/2)
-3. table=60, priority=80,ip,nw_dst=10.0.10.0/24 actions=conjunction(5,2)
-4. table=60, priority=80,conj_id=5,ip actions=load:0x3->NXM_NX_REG5[],load:0x1->NXM_NX_REG0[20],resubmit(,61)
+1. table=EndpointDNAT, priority=200,reg0=0x4000/0x4000 actions=controller(reason=no_match,id=62373,userdata=04)
+2. table=EndpointDNAT, priority=200,tcp,reg3=0xa0a0018,reg4=0x20050/0x7ffff actions=ct(commit,table=AntreaPolicyEgressRule,zone=65520,nat(dst=10.10.0.24:80),exec(set_field:0x10/0x10->ct_mark,move:NXM_NX_REG0[0..3]->NXM_NX_CT_MARK[0..3]))
+3. table=EndpointDNAT, priority=200,tcp,reg3=0xa0a0106,reg4=0x20050/0x7ffff actions=ct(commit,table=AntreaPolicyEgressRule,zone=65520,nat(dst=10.10.1.6:80),exec(set_field:0x10/0x10->ct_mark,move:NXM_NX_REG0[0..3]->NXM_NX_CT_MARK[0..3]))
+4. table=EndpointDNAT, priority=190,reg4=0x20000/0x70000 actions=set_field:0x10000/0x70000->reg4,resubmit(,ServiceLB)
+5. table=EndpointDNAT, priority=0 actions=goto_table:AntreaPolicyEgressRule
 ```
 
-The table-miss flow entry, which is used for non-isolated Pods, forwards
-traffic to the next table EgressMetricsTable, then ([L3ForwardingTable]).
+Flow 1 is designed for Services without Endpoints. It identifies the first packet of connections destined for such Service
+by matching `SvcNoEpRegMark`. Subsequently, the packet is forwarded to the OpenFlow controller (Antrea Agent) for
+further processing.
```

Comment:
I think it wouldn't hurt to indicate what "further processing" is here: for TCP Service traffic we will send a TCP RST, and for all other cases we will send an ICMP Destination Unreachable message.

Reply:
Done, second to you on this point.